### PR TITLE
London Class 8 - Jacint M David - JavaScript Core 2 - Week 3

### PR DESCRIPTION
**Pull request by jacint-david**
@gixo please see my unfinished assignment for this week's coursework.

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-exercises/C-highlight-words/script.js [hunk 1]
```diff
@@ -1,5 +1,28 @@
 function highlightWords(paragraph, colours) {
   // Write your code here...
+  const PARAGRAPH = document.createElement("p"),
+    SELECT = document.createElement("select"),
+    WORDS = paragraph.split(" ");
```
Comment (gixo) [inline 1-exercises/C-highlight-words/script.js:5]: No need for an upper casing here, as it's not an enumerator or static property.

File: 1-exercises/A-reading-list/script.js [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+    newParagraph.innerText = `${book.title} by ${book.author}`;
+    let newImg = document.createElement("img");
+    newImg.src = `${book.bookCoverImage}`;
+    let newLi = document.createElement("li");
+    book.alreadyRead
+      ? newLi.classList.add("green")
+      : newLi.classList.add("red");
+    newLi.append(newParagraph, newImg);
+    newUl.append(newLi);
+  }
```
Comment (gixo) [inline 1-exercises/A-reading-list/script.js:16]: Well done here, I appreciate the use of native DOM manipulation APIs rather than using innerHTML or equivalent methods.

File: 1-exercises/B-todo-list/script.js [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+  for (let todo of todos) {
+    let li = document.createElement("li");
+    li.innerText = `${todo.todo}`;
+    li.addEventListener("click", () => {
+      if (li.classList.contains("done")) {
+        li.classList.remove("done");
+      } else {
+        li.classList.add("done");
+      }
+    });
```
Comment (gixo) [inline 1-exercises/B-todo-list/script.js:14]: The code is functionally correct, but it would have been better to abstract this arrow function into its own named function instead.

File: 1-exercises/B-todo-list/script.js [hunk 4]
```diff
@@ -1,12 +1,26 @@
...
+  const UL = document.createElement("ul");
+  document.getElementById("content").append(UL);
+  for (let todo of todos) {
+    let li = document.createElement("li");
+    li.innerText = `${todo.todo}`;
+    li.addEventListener("click", () => {
+      if (li.classList.contains("done")) {
+        li.classList.remove("done");
+      } else {
+        li.classList.add("done");
```
Comment (gixo) [inline 1-exercises/B-todo-list/script.js:12]: Instead of the 'add' and 'remove' methods, you can use the 'toggle' method to switch a class on and off. See more info here: https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle

File: 1-exercises/C-highlight-words/script.js [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
+  for (let word of WORDS) {
+    let span = document.createElement("span");
+    span.innerText = `${word} `;
+    span.addEventListener("click", () => {
+      if (SELECT.value === "none") {
+        span.style.backgroundColor = "";
+      } else {
+        span.style.backgroundColor = SELECT.value;
+      }
+    });
```
Comment (gixo) [inline 1-exercises/C-highlight-words/script.js:21]: Same suggestion as earlier: better to isolate this into a named function rather than using an arrow function.

**Review by gixo — COMMENTED** [review]
Well done on the work completed so far! I only raised a few smaller points. Keep going :)

File: 1-exercises/B-todo-list/script.js (hunk 4 above)
Comment (jacint-david) [reply 1-exercises/B-todo-list/script.js:12]: Thanks @gixo ! That's super useful to know! 👍

File: 1-exercises/C-highlight-words/script.js (hunk 1 above)
Comment (jacint-david) [reply 1-exercises/C-highlight-words/script.js:5]: oh I thought the general rule of thumb was to always use a const if you don't plan on reassigning the value and if you do use a const you should use all caps. It's good to know that that isn't the case though. I'll google what an enumerator and a static property is.

File: 1-exercises/A-reading-list/script.js (hunk 2 above)
Comment (jacint-david) [reply 1-exercises/A-reading-list/script.js:16]: I read somewhere that innerHTML is dangerous security wise, so I'm trying to avoid it as best as possible even if it isn't on an input.

File: 1-exercises/C-highlight-words/script.js (hunk 1 above)
Comment (gixo) [reply 1-exercises/C-highlight-words/script.js:5]: Every style guide has different recommendations on casing, see for example the Airbnb style guide on casing here: https://github.com/airbnb/javascript#naming--uppercase

Typically, you will only want uppercase casing when a variable is constant and also static i.e. it won't change in the different runs of your program. Think of an `API_KEY` for example.

File: 2-projects/1-todo-list/script.js [hunk 6]
```diff
@@ -1,6 +1,45 @@
...
+      "d-flex",
+      "justify-content-between",
+      "align-items-center"
+    );
+    let span = document.createElement("span"),
+      icon1 = document.createElement("i"),
+      icon2 = document.createElement("i");
+    span.classList.add("badge", "bg-primary", "rounded-pill");
+    icon1.classList.add("fa", "fa-check");
+    icon1.style.marginRight = "0.25rem";
```
Comment (gixo) [inline 2-projects/1-todo-list/script.js:19]: Try to avoid setting inline styles, but instead rely on CSS classes only.

File: 2-projects/1-todo-list/script.js [hunk 7]
```diff
@@ -1,6 +1,45 @@
...
+    icon1.addEventListener("click", () => {
+      if (
+        icon1.parentElement.parentElement.style.textDecoration ===
+        "line-through"
+      ) {
+        icon1.parentElement.parentElement.style.textDecoration = "none";
+      } else {
+        icon1.parentElement.parentElement.style.textDecoration = "line-through";
+      }
+    });
```
Comment (gixo) [inline 2-projects/1-todo-list/script.js:35]: The code is quite complex and hard to follow, I'd suggest breaking it down into several distinct functions.

**Review by gixo — CHANGES_REQUESTED** [review]
Keep working on the code, looking good so far! I'd encourage you to break down the more complex logic into several functions to increase legibility.
